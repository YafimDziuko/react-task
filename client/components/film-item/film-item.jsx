import React from 'react';
import PropTypes from 'prop-types';

import './film-item.scss';

export const FilmItem = (props) => (
    <div className="film-item">
        <img src={props.posterUrl} />
        <div className="film-info">
            <div>
                <p className="title">{props.title}</p>
                <p className="genre">{props.genre}</p>
            </div>
            <p className="year">{props.year}</p>
        </div>
    </div>
)

FilmItem.propTypes = {
    title: PropTypes.string,
    posterUrl: PropTypes.string,
    genre: PropTypes.string,
    year: PropTypes.number,
};

FilmItem.defaultProps = {
    title: "Wolf wood",
    posterUrl: "http://bipbap.ru/wp-content/uploads/2017/05/VOLKI-krasivye-i-ochen-umnye-zhivotnye.jpg",
    genre: "horror",
    year: 1992,
};