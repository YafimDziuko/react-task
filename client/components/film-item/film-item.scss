.film-item {
    width: 33%;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-bottom: 15px;
    

    img {
        width: 150px;
        height: 250px;
    }

    .film-info {
        width: 140px;
        display: flex;
        flex-direction: row;
        align-items: flex-start;
        padding: 5px;
        margin: 0;
        font-weight: 700;
        
        div {
            width: 100px;
            margin-right: 10px;

            .title {
                margin-bottom: 8px;
                font-size: 12px;
            }
    
            .genre {
                font-size: 9px;
            }
        }

        .year {
            width: 30px;
            padding: 2px 4px;
            border: 1px solid black;
            font-size: 10px;
            flex: 1;
        }
    }
}