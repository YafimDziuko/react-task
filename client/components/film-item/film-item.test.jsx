import React from 'react';
import { FilmItem } from './film-item';

const testData = {
  title: "Title",
  posterUrl: "PosterUrl",
  genre: "Genre",
  year: 1900
}

describe('<FilmItem/>', function() {
  it('should render Film Item and match snapshot', function() {
      const wrapper = shallow(<FilmItem />);
      expect(wrapper).toMatchSnapshot();
  });

  it('should render Film Item and match snapshot with props', function() {
    const wrapper = shallow(<FilmItem props={testData}/>);
    expect(wrapper).toMatchSnapshot();
});
});