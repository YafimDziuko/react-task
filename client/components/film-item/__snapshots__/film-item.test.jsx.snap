// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmItem/> should render Film Item and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilmItem
    fullLoad={[Function]}
    props={
      Object {
        "genres": Array [
          "Genre",
        ],
        "id": 123456,
        "match": Object {
          "params": Object {
            "id": 123456,
          },
        },
        "posterUrl": "PosterUrl",
        "title": "Title",
        "year": 1900,
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    user={
      Object {
        "fullItem": Object {
          "isActive": false,
        },
        "inputValue": "",
        "results": Object {
          "data": Array [],
        },
        "resultsCount": -1,
        "search": "title",
        "searchInfo": Object {},
        "sort": "releaseDate",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          src="test-file-stub"
        />,
        <div
          className="film-info"
        >
          <div>
            <Link
              onClick={[Function]}
              replace={false}
              to="/film/undefined"
            >
              <p
                className="title"
              />
            </Link>
            <p
              className="genre"
            />
          </div>
          <p
            className="year"
          />
        </div>,
      ],
      "className": "film-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "test-file-stub",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <Link
                onClick={[Function]}
                replace={false}
                to="/film/undefined"
              >
                <p
                  className="title"
                />
              </Link>
              <p
                className="genre"
              />
            </div>,
            <p
              className="year"
            />,
          ],
          "className": "film-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  onClick={[Function]}
                  replace={false}
                  to="/film/undefined"
                >
                  <p
                    className="title"
                  />
                </Link>,
                <p
                  className="genre"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <p
                    className="title"
                  />,
                  "onClick": [Function],
                  "replace": false,
                  "to": "/film/undefined",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "genre",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "year",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            src="test-file-stub"
          />,
          <div
            className="film-info"
          >
            <div>
              <Link
                onClick={[Function]}
                replace={false}
                to="/film/undefined"
              >
                <p
                  className="title"
                />
              </Link>
              <p
                className="genre"
              />
            </div>
            <p
              className="year"
            />
          </div>,
        ],
        "className": "film-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "test-file-stub",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <Link
                  onClick={[Function]}
                  replace={false}
                  to="/film/undefined"
                >
                  <p
                    className="title"
                  />
                </Link>
                <p
                  className="genre"
                />
              </div>,
              <p
                className="year"
              />,
            ],
            "className": "film-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    onClick={[Function]}
                    replace={false}
                    to="/film/undefined"
                  >
                    <p
                      className="title"
                    />
                  </Link>,
                  <p
                    className="genre"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <p
                      className="title"
                    />,
                    "onClick": [Function],
                    "replace": false,
                    "to": "/film/undefined",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "title",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "genre",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "year",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<FilmItem/> should render Film Item without poster url and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilmItem
    fullLoad={[Function]}
    props={
      Object {
        "genres": Array [
          "Genre",
        ],
        "id": 123456,
        "match": Object {
          "params": Object {
            "id": 123456,
          },
        },
        "title": "Title",
        "year": 1900,
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    user={
      Object {
        "fullItem": Object {
          "isActive": false,
        },
        "inputValue": "",
        "results": Object {
          "data": Array [],
        },
        "resultsCount": -1,
        "search": "title",
        "searchInfo": Object {},
        "sort": "releaseDate",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          src="test-file-stub"
        />,
        <div
          className="film-info"
        >
          <div>
            <Link
              onClick={[Function]}
              replace={false}
              to="/film/undefined"
            >
              <p
                className="title"
              />
            </Link>
            <p
              className="genre"
            />
          </div>
          <p
            className="year"
          />
        </div>,
      ],
      "className": "film-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "test-file-stub",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <Link
                onClick={[Function]}
                replace={false}
                to="/film/undefined"
              >
                <p
                  className="title"
                />
              </Link>
              <p
                className="genre"
              />
            </div>,
            <p
              className="year"
            />,
          ],
          "className": "film-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  onClick={[Function]}
                  replace={false}
                  to="/film/undefined"
                >
                  <p
                    className="title"
                  />
                </Link>,
                <p
                  className="genre"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <p
                    className="title"
                  />,
                  "onClick": [Function],
                  "replace": false,
                  "to": "/film/undefined",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "genre",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "year",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            src="test-file-stub"
          />,
          <div
            className="film-info"
          >
            <div>
              <Link
                onClick={[Function]}
                replace={false}
                to="/film/undefined"
              >
                <p
                  className="title"
                />
              </Link>
              <p
                className="genre"
              />
            </div>
            <p
              className="year"
            />
          </div>,
        ],
        "className": "film-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "test-file-stub",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <Link
                  onClick={[Function]}
                  replace={false}
                  to="/film/undefined"
                >
                  <p
                    className="title"
                  />
                </Link>
                <p
                  className="genre"
                />
              </div>,
              <p
                className="year"
              />,
            ],
            "className": "film-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    onClick={[Function]}
                    replace={false}
                    to="/film/undefined"
                  >
                    <p
                      className="title"
                    />
                  </Link>,
                  <p
                    className="genre"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <p
                      className="title"
                    />,
                    "onClick": [Function],
                    "replace": false,
                    "to": "/film/undefined",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "title",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "genre",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "year",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
